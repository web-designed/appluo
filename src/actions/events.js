import uuid from 'uuid'
import moment from 'moment'

//*******************************************************
// EVENTS ACTION GENERATORS
//*******************************************************

const addEvent = (
   {
      note = '', 
      cleaner = 'unknown', 
      cleanedAt = moment().valueOf(),
      createdAt = moment().valueOf(),
      place = '',
      comments = [],
      id = uuid()
   } = {}) => ({
      type:'ADD_EVENT',
      event: {
         id,
         note,
         cleaner,
         place,
         createdAt,
         cleanedAt,
         comments
      }
   }
)

const removeEvent = (id) => ({
   type: 'REMOVE_EVENT',
   id
})

const editEvent = (id, update) => ({
   type: 'EDIT_EVENT', 
   id,
   update
})

const addComment = (eventId, { commentBody, commenter }) => ({
   type: 'ADD_COMMENT',
   eventId,
   comment: {
      id: uuid(),
      createdAt: moment().valueOf(),
      commenter,
      commentBody
   }
})

const removeComment = (eventId, commentId) => ({
   type:'REMOVE_COMMENT',
   eventId,
   commentId
})

const editComment = (eventId, commentId, update) => ({
   type:'EDIT_COMMENT',
   eventId,
   commentId,
   update
})

export { addEvent, removeEvent, editEvent, addComment, removeComment, editComment }