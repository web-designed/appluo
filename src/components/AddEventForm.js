import React from 'react'
import { connect } from 'react-redux'
import uuid from 'uuid'
import moment from 'moment'
import 'react-dates/initialize'
import { SingleDatePicker } from 'react-dates'
import { Link } from 'react-router-dom'

export class AddEventForm extends React.Component {

   state = {
      date: this.props.cleanedAt ? moment(this.props.cleanedAt) : moment(),
      focused: false,
      note: this.props.note ? this.props.note : '',
      place: this.props.place ? this.props.place : this.props.settings.places[0],
      error: '',
      cleaner: this.props.cleaner ? this.props.cleaner : this.props.settings.cleaners[0],
   }

   onCleanerChange = (e) => {
      const cleaner = e.target.value
      this.setState(() => ({ cleaner }))
   }

   onPlaceChange = (e) => {
      const place = e.target.value
      this.setState(() => ({ place }))
   }

   onNoteChange = (e) => {
      const note = e.target.value
      this.setState(() => ({ note }))
   }

   onDateChange = (date) => {
      this.setState(() => ({ date }))
   }

   onFocusChange = ({focused}) => {
      this.setState(() => ({focused}))
   }

   handleSubmit = (e) => {

      e.preventDefault()
      
      const event = {
         cleanedAt: this.state.date.valueOf(),
         createdAt: moment().valueOf(),
         cleaner: this.state.cleaner,
         place: this.state.place,
         note: this.state.note,
      }

      this.props.handleSubmit(event)
   }

   render(){

      console.log(this.props)

      return(
         <div>
            <form onSubmit={this.handleSubmit}>
               <div class="input-group mb-2">
                  <div class="input-group-prepend">
                     <label class="input-group-text" for="cleaner">Name</label>
                  </div>
                  <select 
                     class="custom-select" 
                     id="cleaner"
                     name="cleaner" 
                     onChange={this.onCleanerChange}
                     value={this.state.cleaner.toLowerCase()}
                  >
                     {
                        this.props.settings.cleaners.map( cleaner => {
                           return <option value={cleaner.toLowerCase()} key={cleaner.toLowerCase()}>{cleaner}</option>
                        })
                     }
                  </select>
               </div>
               <div class="input-group mb-2">
                  <div class="input-group-prepend">
                     <label class="input-group-text" for="place">Place</label>
                  </div>
                  <select 
                     class="custom-select" 
                     id="place"
                     name="place" 
                     onChange={this.onPlaceChange}
                     value={this.state.place.toLowerCase()}
                  >
                     {
                        this.props.settings.places.map( place => {
                           return <option value={place.toLowerCase()} key={place}>{place}</option>
                        })
                     }
                  </select>
               </div>
               <div class="input-group mb-2">
                  <div class="input-group-prepend">
                     <label class="input-group-text" for="date">Date</label>
                  </div>
                  <SingleDatePicker 
                     date={this.state.date}
                     onDateChange={this.onDateChange}
                     focused={this.state.focused}
                     onFocusChange={this.onFocusChange}
                     numberOfMonths={1}
                     isOutsideRange={() => false}
                  />
               </div>
               <div class="form-group">
                  <textarea 
                     class="form-control"
                     value={this.state.note}
                     name="note" 
                     placeholder="Add your note"
                     onChange={this.onNoteChange}
                  ></textarea>
               </div>
               <div class="form-group text-right">
                  <Link class="text-danger mr-2" to={'/'}>Cancel</Link>
                  <button class="btn btn-primary">Submit</button>
               </div>
            </form>
         </div>
      )
   }
}

const mapStateToProps = ({ settings }) => ({
   settings: settings
})

export default connect(mapStateToProps)(AddEventForm)