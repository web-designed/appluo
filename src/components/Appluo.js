import React from 'react'
import EventsList from './EventsList'

class Appluo extends React.Component {
   render() {
      return (
         <div>
            <EventsList />
         </div>
      )
   }
}

export default Appluo