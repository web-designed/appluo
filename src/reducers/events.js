//*******************************************************
// EVENTS REDUCERS
//*******************************************************

   const eventsReducerDefaultState = []
   const eventsReducer = (state = eventsReducerDefaultState, action) => {
      switch(action.type){
         case 'ADD_EVENT':
            return state.concat(action.event)
         case 'REMOVE_EVENT':
            return state.filter(({ id }) => id !== action.id)
         case 'EDIT_EVENT':
            return state.map((event) => {
               if(event.id === action.id) {
                  return {
                     ...event,
                     ...action.update
                  }
               } else {
                  return event
               }
            })
         case 'ADD_COMMENT':
            return state.map((event) => {
               if(event.id === action.eventId){
                  return {
                     ...event,
                     comments: [...event.comments, action.comment]
                  }
               } else {
                  return event
               }
            })
         case 'REMOVE_COMMENT':
            return state.map((event) => {
               if(event.id === action.eventId){
                  return {
                     ...event,
                     comments: event.comments.filter((comment) => {
                        return comment.id !== action.commentId
                     })
                  }
               } else {
                  return event
               }
            })
         case 'EDIT_COMMENT':
            return state.map((event) => {
               if(event.id === action.eventId){
                  return {
                     ...event,
                     comments: event.comments.map((comment) => {
                        if(comment.id === action.commentId){
                           return {
                              ...comment,
                              ...action.update
                           }
                        } else {
                           return comment
                        }
                     }) 
                  }
               } else {
                  return event
               }
            })
         default:
            return state
      }
   }

export default eventsReducer