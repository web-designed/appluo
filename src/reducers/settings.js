import moment from 'moment'

//*******************************************************
// SETTINGS REDUCER
//*******************************************************

   const settingsReducerDefaultState = {
      wgName: 'The Cleaning Team',
      slogan: 'Manage your cleaning plan',
      places: ['Kitchen', 'Bathroom', 'Saloon'],
      cleaners: ['John', 'Alex', 'Tom', 'Steve'],
      createdAt: moment('2018-12-01'),
      notifications: {
         event: {
            afterDelete: 'The cleaning was successfully removed',
            notFound: "The cleaning doesn't exist or has been removed"
         },
         comment: {
            validate: 'Please add your comment'
         }
      }
   }
   const settingsReducer = (state = settingsReducerDefaultState, action) => {
      switch(action.type){
         default:
            return state
      }
   } 

   export default settingsReducer