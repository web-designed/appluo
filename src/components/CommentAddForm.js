import React from 'react'
import uuid from 'uuid'
import { connect } from 'react-redux'
import { startAddComment } from '../actions/comments'
import moment from 'moment'

export class CommentAddForm extends React.Component {

   state = {
      error: '',
      commentBody: '',
      commenter: 'kylu'
   }

   onCommentBodyChange = (e) => {
      const commentBody = e.target.value
      this.setState(() => ({ commentBody }))
   }

   handleSubmitComment = (e) => {

      e.preventDefault()

      if(!this.state.commentBody){
         this.setState(() => ({
            error: this.props.notifications.validate
         }))
      } else {
         const newComment = {
            commentBody: this.state.commentBody,
            commenter: this.state.commenter
         }
         //dispatch the action
         this.props.startAddComment(this.props.currentEventId, newComment)

         //clear the comment and the validation error
         this.setState(() => ({
            commentBody: '',
            error: ''
         }))
      }
   }

   render(){
      return (
         <div class="pt-3">
            <form onSubmit={this.handleSubmitComment}>
               <div class="form-group">
                  <textarea 
                     class="form-control bg-light"
                     name="commentBody"
                     value={this.state.commentBody}
                     onChange={this.onCommentBodyChange}
                     placeholder="Add your comment"
                  ></textarea>
                  {this.state.error && <p class="text-danger"><small>{this.state.error}</small></p>}
               </div>
               <div class="form-group">
                  <button class="btn btn-secondary">Add Comment</button>
               </div>
            </form>
         </div>
      )
   }
}

const mapDispatchToProps = dispatch => ({
   startAddComment: (id, newComment) => dispatch(startAddComment(id, newComment))
})

const mapStateToProps = state => ({
   notifications: state.settings.notifications.comment
})

export default connect(mapStateToProps, mapDispatchToProps)(CommentAddForm)