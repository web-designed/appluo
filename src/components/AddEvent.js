import React from 'react'
import AddEventForm from './AddEventForm'

const AddEvent = () => {
   return(
      <div>
         <h1>Add a cleaning event</h1>
         <AddEventForm />
      </div>
   )
}

export default AddEvent