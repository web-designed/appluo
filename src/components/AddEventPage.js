import React from 'react'
import AddEventForm from './AddEventForm'
import { addEvent } from '../actions/events'
import { connect } from 'react-redux'

const AddEventPage = (props) => (
   <div>
      <h1>Add a cleaning</h1>
      <AddEventForm 
         handleSubmit={ (event) => {
            props.onSubmit(event)
            props.history.push('/')
         }} 
         history={props.history} 
      />
   </div>
)

const mapDispatchToProps = (dispatch) => ({
   onSubmit: (expense) => dispatch(addEvent(expense))
})

export default connect(undefined, mapDispatchToProps)(AddEventPage)