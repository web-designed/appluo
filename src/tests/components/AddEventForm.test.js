import React from 'react'
import { AddEventForm } from '../../components/AddEventForm'
import { shallow } from 'enzyme'
import settings from '../fixtures/settings'
import events from '../fixtures/events'

test('should render the AddEventForm for the Add Event page correctly', () => {
   const wrapper = shallow(<AddEventForm settings={settings} />)
   expect(wrapper).toMatchSnapshot()
})

test('should render the AddEventForm for the Edit Event page correctly', () => {
   const wrapper = shallow(<AddEventForm settings={settings} {...events[0]} />)
   expect(wrapper).toMatchSnapshot()
})