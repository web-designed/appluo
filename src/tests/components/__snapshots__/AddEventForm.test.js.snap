// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the AddEventForm for the Add Event page correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <p>
      <label
        htmlFor="cleaner"
      >
        Name
      </label>
      <select
        name="cleaner"
        onChange={[Function]}
        value="kylu"
      >
        <option
          key="All"
          value="all"
        >
          All
        </option>
        <option
          key="Kylu"
          value="kylu"
        >
          Kylu
        </option>
        <option
          key="Alex"
          value="alex"
        >
          Alex
        </option>
        <option
          key="Till"
          value="till"
        >
          Till
        </option>
        <option
          key="Chris"
          value="chris"
        >
          Chris
        </option>
      </select>
    </p>
    <p>
      <label
        htmlFor="place"
      >
        Wo
      </label>
      <select
        name="place"
        onChange={[Function]}
        value="bad"
      >
        <option
          key="All"
          value="all"
        >
          All
        </option>
        <option
          key="Küche"
          value="küche"
        >
          Küche
        </option>
        <option
          key="Bad"
          value="bad"
        >
          Bad
        </option>
        <option
          key="Flur"
          value="flur"
        >
          Flur
        </option>
      </select>
    </p>
    <p>
      <withStyles(SingleDatePicker)
        anchorDirection="left"
        appendToBody={false}
        block={false}
        calendarInfoPosition="bottom"
        customCloseIcon={null}
        customInputIcon={null}
        date={"2019-02-08T12:45:58.689Z"}
        daySize={39}
        disableScroll={false}
        disabled={false}
        displayFormat={[Function]}
        enableOutsideDays={false}
        firstDayOfWeek={null}
        focused={false}
        hideKeyboardShortcutsPanel={false}
        horizontalMargin={0}
        horizontalMonthPadding={13}
        id="date"
        initialVisibleMonth={null}
        inputIconPosition="before"
        isDayBlocked={[Function]}
        isDayHighlighted={[Function]}
        isOutsideRange={[Function]}
        isRTL={false}
        keepFocusOnInput={false}
        keepOpenOnDateSelect={false}
        monthFormat="MMMM YYYY"
        navNext={null}
        navPrev={null}
        noBorder={false}
        numberOfMonths={1}
        onClose={[Function]}
        onDateChange={[Function]}
        onFocusChange={[Function]}
        onNextMonthClick={[Function]}
        onPrevMonthClick={[Function]}
        openDirection="down"
        orientation="horizontal"
        phrases={
          Object {
            "calendarLabel": "Calendar",
            "chooseAvailableDate": [Function],
            "clearDate": "Clear Date",
            "closeDatePicker": "Close",
            "dateIsSelected": [Function],
            "dateIsUnavailable": [Function],
            "enterKey": "Enter key",
            "escape": "Escape key",
            "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
            "homeEnd": "Home and end keys",
            "jumpToNextMonth": "Move forward to switch to the next month.",
            "jumpToPrevMonth": "Move backward to switch to the previous month.",
            "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
            "keyboardShortcuts": "Keyboard Shortcuts",
            "leftArrowRightArrow": "Right and left arrow keys",
            "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
            "moveFocusByOneMonth": "Switch months.",
            "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
            "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
            "openThisPanel": "Open this panel.",
            "pageUpPageDown": "page up and page down keys",
            "questionMark": "Question mark",
            "returnFocusToInput": "Return to the date input field.",
            "selectFocusedDate": "Select the date in focus.",
            "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
            "upArrowDownArrow": "up and down arrow keys",
          }
        }
        placeholder="Date"
        readOnly={false}
        regular={false}
        renderCalendarInfo={null}
        renderDayContents={null}
        renderMonthElement={null}
        renderMonthText={null}
        reopenPickerOnClearDate={false}
        required={false}
        screenReaderInputMessage=""
        showClearDate={false}
        showDefaultInputIcon={false}
        small={false}
        verticalHeight={null}
        verticalSpacing={22}
        weekDayFormat="dd"
        withFullScreenPortal={false}
        withPortal={false}
      />
    </p>
    <p>
      <label
        htmlFor="note"
      >
        Note
      </label>
      <textarea
        name="note"
        onChange={[Function]}
        placeholder="zB: Alles sauber gemacht"
        value=""
      />
    </p>
    <p>
      <button>
        Submit
      </button>
    </p>
  </form>
</div>
`;

exports[`should render the AddEventForm for the Edit Event page correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <p>
      <label
        htmlFor="cleaner"
      >
        Name
      </label>
      <select
        name="cleaner"
        onChange={[Function]}
        value="Kylu"
      >
        <option
          key="All"
          value="all"
        >
          All
        </option>
        <option
          key="Kylu"
          value="kylu"
        >
          Kylu
        </option>
        <option
          key="Alex"
          value="alex"
        >
          Alex
        </option>
        <option
          key="Till"
          value="till"
        >
          Till
        </option>
        <option
          key="Chris"
          value="chris"
        >
          Chris
        </option>
      </select>
    </p>
    <p>
      <label
        htmlFor="place"
      >
        Wo
      </label>
      <select
        name="place"
        onChange={[Function]}
        value="bad"
      >
        <option
          key="All"
          value="all"
        >
          All
        </option>
        <option
          key="Küche"
          value="küche"
        >
          Küche
        </option>
        <option
          key="Bad"
          value="bad"
        >
          Bad
        </option>
        <option
          key="Flur"
          value="flur"
        >
          Flur
        </option>
      </select>
    </p>
    <p>
      <withStyles(SingleDatePicker)
        anchorDirection="left"
        appendToBody={false}
        block={false}
        calendarInfoPosition="bottom"
        customCloseIcon={null}
        customInputIcon={null}
        date={"2019-02-08T14:07:11.467Z"}
        daySize={39}
        disableScroll={false}
        disabled={false}
        displayFormat={[Function]}
        enableOutsideDays={false}
        firstDayOfWeek={null}
        focused={false}
        hideKeyboardShortcutsPanel={false}
        horizontalMargin={0}
        horizontalMonthPadding={13}
        id="date"
        initialVisibleMonth={null}
        inputIconPosition="before"
        isDayBlocked={[Function]}
        isDayHighlighted={[Function]}
        isOutsideRange={[Function]}
        isRTL={false}
        keepFocusOnInput={false}
        keepOpenOnDateSelect={false}
        monthFormat="MMMM YYYY"
        navNext={null}
        navPrev={null}
        noBorder={false}
        numberOfMonths={1}
        onClose={[Function]}
        onDateChange={[Function]}
        onFocusChange={[Function]}
        onNextMonthClick={[Function]}
        onPrevMonthClick={[Function]}
        openDirection="down"
        orientation="horizontal"
        phrases={
          Object {
            "calendarLabel": "Calendar",
            "chooseAvailableDate": [Function],
            "clearDate": "Clear Date",
            "closeDatePicker": "Close",
            "dateIsSelected": [Function],
            "dateIsUnavailable": [Function],
            "enterKey": "Enter key",
            "escape": "Escape key",
            "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
            "homeEnd": "Home and end keys",
            "jumpToNextMonth": "Move forward to switch to the next month.",
            "jumpToPrevMonth": "Move backward to switch to the previous month.",
            "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
            "keyboardShortcuts": "Keyboard Shortcuts",
            "leftArrowRightArrow": "Right and left arrow keys",
            "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
            "moveFocusByOneMonth": "Switch months.",
            "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
            "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
            "openThisPanel": "Open this panel.",
            "pageUpPageDown": "page up and page down keys",
            "questionMark": "Question mark",
            "returnFocusToInput": "Return to the date input field.",
            "selectFocusedDate": "Select the date in focus.",
            "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
            "upArrowDownArrow": "up and down arrow keys",
          }
        }
        placeholder="Date"
        readOnly={false}
        regular={false}
        renderCalendarInfo={null}
        renderDayContents={null}
        renderMonthElement={null}
        renderMonthText={null}
        reopenPickerOnClearDate={false}
        required={false}
        screenReaderInputMessage=""
        showClearDate={false}
        showDefaultInputIcon={false}
        small={false}
        verticalHeight={null}
        verticalSpacing={22}
        weekDayFormat="dd"
        withFullScreenPortal={false}
        withPortal={false}
      />
    </p>
    <p>
      <label
        htmlFor="note"
      >
        Note
      </label>
      <textarea
        name="note"
        onChange={[Function]}
        placeholder="zB: Alles sauber gemacht"
        value="Alles sauber gemacht"
      />
    </p>
    <p>
      <button>
        Submit
      </button>
    </p>
  </form>
</div>
`;
